src/Features/LanguageServer/Protocol/Features/Diagnostics/DiagnosticAnalyzerService.cs

```diff
@@ -94,23 +89,20 @@ public void Reanalyze(Workspace workspace, IEnumerable<ProjectId>? projectIds, I
             bool isExplicit,
             CancellationToken cancellationToken)
         {
-            if (_map.TryGetValue(document.Project.Solution.Workspace, out var analyzer))
+            var analyzer = CreateIncrementalAnalyzer(document.Project.Solution.Workspace);
```

Comment:
all these methods went from bailing out if there was no analyzer (meaning solution-crawler hadn't initialized it) to creating it on demand.  Now everything can be 'pull' and we just do what is needed when the pull request comes through.